src/test/java/com/google/api/codegen/testdata/java_test_mock_service_impl_library.baseline

```diff
@@ -46,154 +47,229 @@ import java.util.List;
-       StreamObserver<Book> responseObserver) {
-     requests.add(request);
-     responseObserver.onNext(Book.newBuilder().build());
-     responseObserver.onCompleted();
-   }
-
-   @Override
-   public void listBooks(ListBooksRequest request,
-       StreamObserver<ListBooksResponse> responseObserver) {
-     requests.add(request);
-     responseObserver.onNext(ListBooksResponse.newBuilder().build());
-     responseObserver.onCompleted();
-   }
-
-   @Override
-   public void deleteBook(DeleteBookRequest request,
-       StreamObserver<Empty> responseObserver) {
-     requests.add(request);
-     responseObserver.onNext(Empty.newBuilder().build());
-     responseObserver.onCompleted();
-   }
-
-   @Override
-   public void updateBook(UpdateBookRequest request,
-       StreamObserver<Book> responseObserver) {
-     requests.add(request);
-     responseObserver.onNext(Book.newBuilder().build());
-     responseObserver.onCompleted();
-   }
-
-   @Override
-   public void moveBook(MoveBookRequest request,
-       StreamObserver<Book> responseObserver) {
-     requests.add(request);
-     responseObserver.onNext(Book.newBuilder().build());
-     responseObserver.onCompleted();
-   }
-
-   @Override
-   public void listStrings(ListStringsRequest request,
-       StreamObserver<ListStringsResponse> responseObserver) {
-     requests.add(request);
-     responseObserver.onNext(ListStringsResponse.newBuilder().build());
-     responseObserver.onCompleted();
-   }
-
-   @Override
-   public void addComments(AddCommentsRequest request,
-       StreamObserver<Empty> responseObserver) {
-     requests.add(request);
-     responseObserver.onNext(Empty.newBuilder().build());
-     responseObserver.onCompleted();
-   }
-
-   @Override
-   public void getBookFromArchive(GetBookFromArchiveRequest request,
-       StreamObserver<Book> responseObserver) {
-     requests.add(request);
-     responseObserver.onNext(Book.newBuilder().build());
-     responseObserver.onCompleted();
-   }
-
-   @Override
-   public void updateBookIndex(UpdateBookIndexRequest request,
-       StreamObserver<Empty> responseObserver) {
-     requests.add(request);
-     responseObserver.onNext(Empty.newBuilder().build());
-     responseObserver.onCompleted();
-   }
-
-   @Override
-   public void streamShelves(ListShelvesRequest request,
-       StreamObserver<ListShelvesResponse> responseObserver) {
-     requests.add(request);
-     responseObserver.onNext(ListShelvesResponse.newBuilder().build());
-     responseObserver.onCompleted();
-   }
+  private ArrayList<GeneratedMessage> requests;
+  private ArrayList<GeneratedMessage> responses;
+
+  public MockLibraryServiceImpl() {
+    requests = new ArrayList<>();
+    responses = new ArrayList<>();
+  }
+
+  public List<GeneratedMessage> getRequests() {
+    return requests;
+  }
+
+  public void setResponses(List<GeneratedMessage> responses) {
+    this.responses = Lists.newArrayList(responses);
+  }
```

Comment:
I would expect to use more responses for the page-streaming case after the response field populating work is done.  Updated the issue: https://github.com/googleapis/toolkit/issues/354